pkg/services/ngalert/state/template.go

```diff
@@ -87,3 +86,20 @@ func newTemplateCaptureValues(values map[string]eval.NumberValueCapture) map[str
 	}
 	return m
 }
+
+func buildExploreLink(showGraph, showTable bool) func(string) string {
+	return func(rawQuery string) string {
+		var query struct {
+			Datasource string `json:"datasource"`
+			Expr       string `json:"expr"`
+		}
+
+		if err := json.Unmarshal([]byte(rawQuery), &query); err != nil {
+			return ""
+		}
+
+		escapedExpression := url.QueryEscape(query.Expr)
+		return fmt.Sprintf(
+			`/explore?left=["now-1h","now",%q,{"datasource":%q,"expr":%q,"instant":%v,"range":%v}]`, query.Datasource, query.Datasource, escapedExpression, showTable, showGraph)
```

Comment:
```suggestion
			`/explore?left=["now-1h","now",%[1]q,{"datasource":%[1]q,"expr":%q,"instant":%v,"range":%v}]`, query.Datasource, escapedExpression, showTable, showGraph)
```

Reply:
Also, if you escape query, I think it makes sense to escape data source as well.